src/react-16.3/ContractData.js

```diff
@@ -0,0 +1,118 @@
+import { drizzleConnect } from 'drizzle-react'
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+/*
+ * Create component.
+ */
+
+class ContractData extends Component {
+  constructor(props) {
+    super(props)
+
+    // Fetch initial value from chain and return cache key for reactive updates.
+    var methodArgs = this.props.methodArgs ? this.props.methodArgs : []
+
+    this.contracts = props.drizzle.contracts
+    this.state = {
+      dataKey: this.contracts[this.props.contract].methods[this.props.method].cacheCall(...methodArgs)
+    };
+
+    // Get the contract ABI
+    const abi = this.contracts[this.props.contract].abi;
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { methodArgs, contract, method } = this.props;
+
+    const didContractChange = contract !== nextProps.contract;
+    const didMethodChange = method !== nextProps.method;
+    const didArgsChange = JSON.stringify(methodArgs) !== JSON.stringify(nextProps.methodArgs)
+
+    if (didContractChange || didMethodChange || didArgsChange) {
+      this.setState({
+        dataKey: this.contracts[nextProps.contract].methods[nextProps.method].cacheCall(...nextProps.methodArgs)
+      })
+    }
+  }
+
+  render() {
+    const { drizzle, drizzleState } = this.props;
+
+    // Contract is not yet intialized.
+    if (!drizzleState.contracts[this.props.contract].initialized) {
+      return (
+        <span>Initializing...</span>
+      )
+    }
+
+    // If the cache key we received earlier isn't in the store yet; the initial value is still being fetched.
+    if (!(this.state.dataKey in drizzleState.contracts[this.props.contract][this.props.method])) {
+      return (
+        <span>Fetching...</span>
+      )
+    }
+
+    // Show a loading spinner for future updates.
+    var pendingSpinner = drizzleState.contracts[this.props.contract].synced ? '' : ' 🔄'
```

Comment:
too many `var`s in this file and the other ones. I'll stop commenting on them now.